libs/ng-utils/src/lib/use-http-image-source/use-http-image-source.pipe.ts

```diff
@@ -0,0 +1,77 @@
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { ChangeDetectorRef, Inject, OnDestroy, Pipe, PipeTransform } from '@angular/core';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { BehaviorSubject, of, Subscription } from 'rxjs';
+import { catchError, distinctUntilChanged, filter, map, switchMap } from 'rxjs/operators';
+
+@Pipe({
+  name: 'useHttpImgSrc',
+  pure: false,
+})
+export class UseHttpImageSourcePipe implements PipeTransform, OnDestroy {
+  private subscription!: Subscription;
+  private loadingImagePath!: string;
+  private errorImagePath!: string;
+  private latestValue!: string | SafeUrl;
+  private transformValue = new BehaviorSubject<string>('');
+
+  constructor(
+    private httpClient: HttpClient,
+    private domSanitizer: DomSanitizer,
+    private cdr: ChangeDetectorRef,
+    @Inject('THIS_DOT_LOADING_IMAGE_PATH') private defaultLoadingImagePath: string,
+    @Inject('THIS_DOT_ERROR_IMAGE_PATH') private defaultErrorImagePath: string
+  ) {}
+
+  transform(
+    imagePath: string,
+    loadingImagePath?: string,
+    errorImagePath?: string
+  ): string | SafeUrl {
+    this.setLoadingAndErrorImagePaths(loadingImagePath, errorImagePath);
+    if (!imagePath) {
+      return this.errorImagePath;
+    }
+    if (!this.subscription) {
+      this.setUpSubscription();
+    }
+    this.transformValue.next(imagePath);
+    return this.latestValue || this.loadingImagePath;
+  }
+
+  ngOnDestroy(): void {
+    this.subscription?.unsubscribe();
```

Comment:
If we initialize the subscription inline above we could get rid of this `?`
```suggestion
    this.subscription.unsubscribe();
```

Reply:
done